dev/reference/best-practices/massive-regions.md

```diff
@@ -0,0 +1,143 @@
+
+This document introduces the workflow of Raftstore (a core module of TiKV), explains why a massive amount of Regions cause performance problem, and offers methods for tuning TiKV performance.
+
+## Raftstore workflow
+
+A TiKV instance has multiple Regions on it. The Raftstore module drives Raft state machines to process messages for Regions. These messages include processing read and write requests on Regions, persisting and replicating Raft logs, and processing Raft heartbeats. However, an increasing number of Regions influence performance of the whole cluster. To explain this, it is necessary to learn the workflow of Raftstore first.
+
+![Raftstore Workflow](/media/best-practices/raft-process.png)
+
+> **Note:**
+>
+> This diagram only illustrates the workflow of Raftstore and does not represent the actual code structure.
+
+From this diagram, requests from TiDB, through the gRPC and storage modules, become the final read and write messages of KV (key-value), and are sent to the corresponding Regions. These messages are not immediately processed but are temporarily stored. Raftstore polls to check if each Region has messages to process. If a Region has messages to process, Raftstore drives the Raft state machine of the Region to process these messages and perform subsequent operations according to the state changes generated from these messages. For example, with a write request, the Raft state machine stores logs into disk and sends logs to other Region replicas; sends heartbeat information to other Region replicas when the heartbeat interval is reached.
+
+## Performance problem
+
+From the diagram of Raftstore workflow, messages for each Region are processed one by one. When Regions are in huge numbers, it takes Raftstore sometime to process the heartbeats of these Regions, which can cause some delay. As a result, some read and write requests are not processed in time. If read and write pressure is high, CPU usage of the Raftstore thread might easily reach the bottleneck, which further increases the delay and affects the performance.
+
+Generally, if the CPU usage of the loaded Raftstore reaches 85% or higher, Raftstore is in busy state and becomes the bottleneck. At the same time, `propose wait duration` can be as high as hundreds of milliseconds.
+
+> **Note:**
+>
+> + For the CPU usage of Raftstore as mentioned above, Raftstore is single-threaded. If Raftstore is multi-threaded, CPU usage threshold (85%) can be increased proportionally.
+> + Because I/O operations exist in the Raftstore thread, CPU usage cannot reach 100%.
+
+### Performance monitoring
+
+You can check the following monitoring metrics in Grafana's *TiKV* panel:
+
++ `Raft store CPU` in the *Thread-CPU* panel
```

Comment:
Generally, use the bold style for words in the UI interface. Please update similar occurrences in this PR. Thanks!